exporter/chronicleexporter/exporter.go

```diff
@@ -0,0 +1,129 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chronicleexporter
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"os"
+
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/google"
+)
+
+const scope = "https://www.googleapis.com/auth/malachite-ingestion"
+
+const baseEndpoint = "https://malachiteingestion-pa.googleapis.com"
+
+const apiTarget = "/v2/unstructuredlogentries:batchCreate"
+
+type chronicleExporter struct {
+	cfg        *Config
+	logger     *zap.Logger
+	httpClient *http.Client
+	marshaler  logMarshaler
+	endpoint   string
+}
+
+func newExporter(cfg *Config, params exporter.CreateSettings) (*chronicleExporter, error) {
+	var creds *google.Credentials
+	var err error
+
+	switch {
+	case cfg.Creds != "":
+		creds, err = google.CredentialsFromJSON(context.Background(), []byte(cfg.CredsFilePath), scope)
+		if err != nil {
+			return nil, fmt.Errorf("obtain credentials from JSON: %w", err)
+		}
+	case cfg.CredsFilePath != "":
+		credsData, err := os.ReadFile(cfg.CredsFilePath)
+		if err != nil {
+			return nil, fmt.Errorf("read credentials file: %w", err)
+		}
+
+		creds, err = google.CredentialsFromJSON(context.Background(), credsData, scope)
+		if err != nil {
+			return nil, fmt.Errorf("obtain credentials from JSON: %w", err)
+		}
+	default:
+		return nil, fmt.Errorf("no credentials provided")
+	}
+
+	// Use the credentials to create an HTTP client
+	httpClient := oauth2.NewClient(context.Background(), creds.TokenSource)
+
+	return &chronicleExporter{
+		endpoint:   buildEndpoint(cfg),
+		cfg:        cfg,
+		logger:     params.Logger,
+		httpClient: httpClient,
+		marshaler:  newMarshaler(*cfg),
+	}, nil
+}
+
+func buildEndpoint(cfg *Config) string {
```

Comment:
Can you add some comments on this to indicate how this works or point to docs? It sounds like it wasn't easy to track down this answer so we should document what we know.